Packages/com.unity.inputsystem/InputSystem/API/Input.cs

```diff
@@ -254,48 +258,224 @@ public static partial class Input
         /// </summary>
         public static IReadOnlyList<Gamepad> gamepads { get; }
 
+#endif
+
+        private static ButtonControl GetButtonControl(Inputs input)
+        {
+            // WHAT THE FCK!? or .. can we do better?
```

Comment:
A minor improvement (for enums in the keyboard range at least) might be to move the key mapping out of here and into a dedicated function. Eg.
`
if (input < Inputs.Key_MAX) {
    return Keyboard.current != null ? Keyboard.current[KEYMAP_FUN(input)] : null;
}
`